pkg/util/collate/bin.go

```diff
@@ -65,6 +65,16 @@ func (*binPaddingCollator) Pattern() WildcardPattern {
 	return &binPattern{}
 }
 
+// binPureCollator match pattern in bytes
+type binPureCollator struct {
```

Comment:
How about modify the behavior of `binPattern` directly? I don't think there is anything depending on the original wrong behavior 🤔. Is there any special reason to keep the original `binPattern`?

Reply:
According to https://github.com/pingcap/tidb/pull/20610, some utf_bin collation needs to use Rune characters.

Reply:
You are right. I realized that `utf8_bin` and `utf8mb4_bin` are using `binPaddingCollator`, and `utf8mb4_0900_bin` is using `binCollator` 🤦 (and also `gbkBinCollator` uses the `binPattern` directly).